### PR TITLE
Fixing numpy test for nanmedian

### LINKED ISSUE #13905: Fix averages_and_variances.test_numpy_nanmedian
|   |   |
|---|---|
|paddle|<a href="https://github.com/unifyai/ivy/actions/runs/6412219892/job/17409128634"></a>
|numpy|<a href="https://github.com/unifyai/ivy/actions/runs/6412219892/job/17409128634"></a>
|jax|<a href="https://github.com/unifyai/ivy/actions/runs/6412219892/job/17409128634"></a>
|tensorflow|<a href="https://github.com/unifyai/ivy/actions/runs/6412219892/job/17409128634"></a>
|torch|<a href="https://github.com/unifyai/ivy/actions/runs/6412219892/job/17409128634"></a>

### PR DESCRIPTION
**Pull request by abdulasiraj**
closes #13905

### CODE REVIEW

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Great work @abdulasiraj ! :+1: 

Just requested some changes and left comments on them :slightly_smiling_face: :+1:

File: ivy/functional/backends/jax/experimental/statistical.py [hunk 1]
```diff
@@ -197,9 +197,14 @@ def nanmedian(
     overwrite_input: bool = False,
     out: Optional[JaxArray] = None,
 ) -> JaxArray:
-    return jnp.nanmedian(
-        input, axis=axis, keepdims=keepdims, overwrite_input=overwrite_input, out=out
+    result = jnp.nanmedian(
```
[the remark this answers is not in the document]
Comment (abdulasiraj) [reply ivy/functional/backends/jax/experimental/statistical.py:200]: I think you mean like this:

if overwrite_input:
 result = jnp.nanmedian(input, axis=axis, keepdims=keepdims, overwrite_input=False, out=None)
 input = result.copy()
 return result
else:
 return jnp.nanmedian(input, axis=axis, keepdims=keepdims, overwrite_input=False, out=None)

But if you notice, there's redundancy in the code above
 `jnp.nanmedian(input, axis=axis, keepdims=keepdims, overwrite_input=False, out=None)`

So, I think it's even better to write like this:
result = jnp.nanmedian(
        input, axis=axis, keepdims=keepdims, overwrite_input=False, out=None
    )
    if overwrite_input:
        input = result.copy()
    return result

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Requested some changes :slightly_smiling_face:

File: ivy/functional/backends/jax/experimental/statistical.py [hunk 2]
```diff
@@ -197,8 +197,13 @@ def nanmedian(
...
     out: Optional[JaxArray] = None,
 ) -> JaxArray:
+    if overwrite_input:
+        copied_input = input.copy()
+        return jnp.nanmedian(
+            copied_input, axis=axis, keepdims=keepdims, overwrite_input=False, out=None
+        )
     return jnp.nanmedian(
-        input, axis=axis, keepdims=keepdims, overwrite_input=overwrite_input, out=out
+        input, axis=axis, keepdims=keepdims, overwrite_input=False, out=None
```
Comment (zaeemansari70) [inline ivy/functional/backends/jax/experimental/statistical.py:206]: You should not change the parameters here, they should be passed like they are being passed :+1:

**Review by zaeemansari70 — CHANGES_REQUESTED** [review]
Requested some changes :slightly_smiling_face:

File: ivy/functional/backends/jax/experimental/statistical.py (hunk 2 above)
Comment (abdulasiraj) [reply ivy/functional/backends/jax/experimental/statistical.py:206]: > You should not change the parameters here, they should be passed like they are being passed :+1: 
> `overwrite_input=overwrite_input, out=out`

Thanks Zaeem. But jax doesn't support overwrite=True and out != None in their function nativly while all other frameworks does.

**Comment by rishabgit** [comment]
Hi @abdulasiraj! You'll have to add another argument to implement a superset over the backends. Check out Julia's comment here - https://github.com/unifyai/ivy/pull/13069#pullrequestreview-1413772549

**Comment by abdulasiraj** [comment]
> Hi @abdulasiraj! You'll have to add another argument to implement a superset over the backends. Check out Julia's comment here - [#13069 (review)](https://github.com/unifyai/ivy/pull/13069#pullrequestreview-1413772549)

Thanks @rishabgit,
I'll see julia's comment for sure.

**Review by zaeemansari70 — APPROVED** [review]
Lgtm! :+1:
